corehq/util/metrics/__init__.py

```diff
@@ -313,6 +314,33 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             metrics_counter(self.failed_name)
 
 
+def limit_domains(domain_name):
+    """Return the domain name if it's among a privileged set that get tagged individually
+
+    Else return __other__.  This is used to limit the number of tag combinations sent to datadog.
+    """
+    if not settings.IS_SAAS_ENVIRONMENT:
+        return domain_name
+    if domain_name and domain_name in _domains_to_tag():
+        return domain_name
+    return '__other__'
+
+
+@quickcache([], timeout=24 * 60 * 60)
+def _domains_to_tag():
+    # Only tag big projects individually
+    # I expect this implementation may need to evolve
```

Comment:
My goal here is to tag domains that we'll likely want to be able to audit individually, definitely open to suggestions for how best to capture that.  What I went with for now (and this is heavily influenced by the USH use-case) is:

* All domains on non-SaaS environments
* Enterprise subscription
* Implementation or Sandbox subscription type

For reference, here's the current number of enterprise subscriptions by service type on prod:
```
    {'service_type': 'INTERNAL', 'total': 1297},
    {'service_type': 'PRODUCT', 'total': 805},
    {'service_type': 'NOT_SET', 'total': 213},
    {'service_type': 'IMPLEMENTATION', 'total': 84},
    {'service_type': 'SANDBOX', 'total': 43},
```

(FWIW there are a total of 66 Implementation or Sandbox domains on staging)